### ISSUE TITLE
focus on svm service and its use

### ISSUE THREAD

**Issue by laszewsk** [cloudmesh-community/cloudmesh.nn#8]
Tamdon, when I look at your code you may make it a bit too complicated. 

a) scikt learn has svm for classification, regression, and outlier detection
b) it is only necessary to implement the svm use in an open api service. E.g. assume your caller does not have scikit learn installed or access to it but can callyoue openapi service that runs this servoce. 
c) thus you can rewrite your nfl analysis (which in my view is just a test if a and b works) without actually writing an openapi service for it, but focussing on calling and wrappring svn in its entire functionality into an open api service. This would give a very clean interface.

Thus the nfl analysis is a normal program using the general svn that is now hosted as a service on the openapi server. There should not be a depepndency in the openapi service on nfl analysis, but you just pushing that data, do the analysis and retrieve the result.

**Comment by laszewsk** [comment]
let us look at https://scikit-learn.org/stable/modules/svm.html

you see SVC, NuSVC and LinearSVC 

so when you follow them you find something like
https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html#sklearn.svm.SVC

No that is the spec for which you should create an openapi in general with paarameters, data upload and analysis, not nfl, you could even make 3 services out of it or possibly have the choice if you do SVC, NuSVC or LinearSVC .

Naturally your SVC service has many parameters as you try to expose its general functionality as function to the user. 

So the focus is not to hos nfl as an openapi, but to host svm as an openapi and then use within your example taht open aboy while creating the same parameter function such as 

OpenapiSVM( .... same parameters as scicit learn SVM ....)

than in your nfl example you can do 2 tests

cms set openapi=True .   # or False

from cloudmesh.Variables import Variables()
openapi = Variables()["openapi"]

if openapi:
    from cloudmesh.service.sklearn.svm import SVC
else:
    from sklearn.svm import SVC

Now write your code for nfl

Naturally your code in cloudmesh.sklearn.svm

calls your openapiservice with requests

This way you have 2 implementations. A one that does the nn locally, and b, one that does the remote call, Now as they have the same parameters, you can measuer the performance ith the same pytest, while using the nfl as example. So in teh forgreound in not tehe nfl example, but the implementation of SVM and its general use while nfl is jusr an example.

**Comment by laszewsk** [comment]
e.g. nfl is not part of the openapi service .... but only svm

**Comment by laszewsk** [comment]
so looking at 
>>> import numpy as np
>>> X = np.array([[-1, -1], [-2, -1], [1, 1], [2, 1]])
>>> y = np.array([1, 1, 2, 2])
>>> from sklearn.svm import SVC
>>> clf = SVC(gamma='auto')
>>> clf.fit(X, y) 
SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='auto', kernel='rbf',
    max_iter=-1, probability=False, random_state=None, shrinking=True,
    tol=0.001, verbose=False)
>>> print(clf.predict([[-0.8, -1]]))
[1]

SVC becomes a "resource" withs lots of parameters

and the methods


decision_function(X) | Evaluates the decision function for the samples in X.
-- | --
fit(X, y[, sample_weight]) | Fit the SVM model according to the given training data.
get_params([deep]) | Get parameters for this estimator.
predict(X) | Perform classification on samples in X.
score(X, y[, sample_weight]) | Returns the mean accuracy on the given test data and labels.
set_params(**params) | Set the parameters of this estimator.

become path

so you do this as open api. Then the only otherthing you need to do is to do the "service" wrapper class as i described above that just calls the open api service insetad of thescikit learn service so we can infoke it. Important is that you naturally have some way of uploading the data

**Comment by laszewsk** [comment]
In this case nfl just becomes an example, and you coudl even with the Variables trick and a 5 line change in the examples listed in https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html#sklearn.svm.SVC

benchmark your work for

https://scikit-learn.org/stable/auto_examples/plot_multilabel.html#sphx-glr-auto-examples-plot-multilabel-py

https://scikit-learn.org/stable/auto_examples/plot_kernel_approximation.html#sphx-glr-auto-examples-plot-kernel-approximation-py

https://scikit-learn.org/stable/auto_examples/applications/plot_face_recognition.html#sphx-glr-auto-examples-applications-plot-face-recognition-py

https://scikit-learn.org/stable/auto_examples/classification/plot_digits_classification.html#sphx-glr-auto-examples-classification-plot-digits-classification-py

https://scikit-learn.org/stable/auto_examples/model_selection/plot_confusion_matrix.html#sphx-glr-auto-examples-model-selection-plot-confusion-matrix-py

**Comment by laszewsk** [comment]
to make the integration in your tests even simpler you can jsut ass a prameter to your svm function wrapper that than decides if you access the openapi service or svm via scikit learn locally

here anotehr example to show how simple this ought to be .... 

```
import numpy as np
import matplotlib.pyplot as plt

from cloudmesh.sklearn import SVM # this is the only change needed

from sklearn import datasets
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix
from sklearn.utils.multiclass import unique_labels

# import some data to play with
iris = datasets.load_iris()
X = iris.data
y = iris.target
class_names = iris.target_names

# Split the data into a training set and a test set
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)

# Run classifier, using a model that is too regularized (C too low) to see
# the impact on the results
classifier = svm.SVC(kernel='linear', C=0.01, openapi=True) # this is the only change needed
y_pred = classifier.fit(X_train, y_train).predict(X_test)
```
